experiment/measurer.py

```diff
@@ -376,7 +402,7 @@ def run_cov_new_units(self):
         self.UNIT_BLACKLIST[self.benchmark] = (
             self.UNIT_BLACKLIST[self.benchmark].union(set(crashing_units)))
 
-    def generate_summary(self, cycle: int, summary_only=True):
+    def generate_summary(self, cycle: int, summary_only=False):
```

Comment:
For recording function and segment coverage (to the detail that this PR is recording), we need the llvm to produce a full fledged coverage summary. With `summary_only=True` the coverage summary will have only high level information like {count, covered, percentage} for functions, instantiations and regions. But we need much more detailed information like segment line, column, filename, function hits and so on.

Reply:
I definitely remember this part which is measurer will current fail if we do summary=True , it needs lots of ram and those 48-96 threads just break in parallel. this is the reason why we do it sequentially at end when summary=True. I dont have a great solution for this memory issue until we break measurer into individual bots (will probably come end of year after bug based benchmarking).

Reply:
By default, [llvm-cov auto-detects an an appropriate number of threads to use](https://llvm.org/docs/CommandGuide/llvm-cov.html). Does `-num-threads=1` help?

Reply:
So that one i did explore before, it is only for html report generation i remember
"Use N threads to write file reports (only applicable when -output-dir is specified)"
Maybe Bharath can verify this with local experiment, start 1 fuzzer against all benchmark and one trial and see measurer top memory usage. 

I am really sorry about these measurer limitations, we want to break it, but we have to do these one by one due to context switching with other things like ClusterFuzz,OSS-Fuzz.

Reply:
No worries, at all. If there is a limitation that we cannot resolve at our end, we'll give up on this PR (and wait for the live coverage stats). First, we'll see if there is an easy workaround, though. We'll come back on the memory consumption with one thread, one trial, one fuzzer, all benchmarks, for two snapshot iterations (like, 1860 seconds).

Reply:
Hi @inferno-chromium, I tried the the configuration that you suggested but against three benchmarks since I'm running fuzzbench locally. The configuration is mentioned below for your reference.

```
Experiment config (local):
        benchmark:  libjpeg-turbo-07-2017,  freetype2-2017,  harfbuzz-1.3.2
        fuzzers:  libfuzzer
        trials:  1
        max_total_time:  1860 (2 cycles)
```

So, with the `-summary-only` flag for the `llvm-cov export` call, the memory consumption by these three threads totalled to 2.43 GB.

On the flip side, without the `-summary-only` flag and with `-num-threads=1`, `-skip-expansions` and `-region-coverage-gt=0` flags the memory consumption totalled to 3.35 GB for these three multiprocess threads. 
Its safe to say that, the `-num-thread=1` option works without an output directory specified. `-skip-expansions` and  `-region-coverage-gt=0` would also help reduce the load on the `llvm-cov export` call when exporting full summary as it trims down all the parts of the JSON output that we do not want. Hope this helps :)

Reply:
-num-threads=1, -skip-expansions are ok, but i think we cant use -region-coverage-gt=0 since we need info on what is not getting covered ? can you also try this on php benchmark, that one i remember was the most memory intensive.
Also, this code is getting heavily refactored by @jonathanmetzman to break up measurer on multiple bot, so i suggest waiting till that work is finished.

Reply:
No worries at all, I'll try this on php benchmark and get back to you on the memory consumption.

With `-region-coverage-gt=0`, I took the liberty of adding it, as `covered_regions.json` records information on regions with hit_count greater than 0. If it cannot be used, I'll remove it, no worries. https://github.com/google/fuzzbench/blob/b9853d50eb96fcd233f2a233457056e4122234fa/experiment/measurer/coverage_utils.py#L289.

Eagerly waiting on the queue based measurer :)

Reply:
On trying the same on `php_php-fuzz-parser` benchmark (only benchmark for the experiment), 
- The memory consumption by the `llvm-cov export` call with the `-summary-only` flag is about **214 MB** .
- llvm-cov export` call without the `-summary-only` flag and with `-num-threads=1`, `-skip-expansions` and `-region-coverage-gt=0` flags the memory consumption is **627 MB**.